modules/pricer/src/main/java/com/opengamma/strata/pricer/capfloor/SabrIborCapletFloorletVolatilityBootstrapper.java

```diff
@@ -0,0 +1,381 @@
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+import com.opengamma.strata.basics.ReferenceData;
+import com.opengamma.strata.basics.currency.Currency;
+import com.opengamma.strata.basics.index.IborIndex;
+import com.opengamma.strata.collect.ArgChecker;
+import com.opengamma.strata.collect.array.DoubleArray;
+import com.opengamma.strata.collect.array.DoubleMatrix;
+import com.opengamma.strata.market.ValueType;
+import com.opengamma.strata.market.curve.Curve;
+import com.opengamma.strata.market.curve.CurveMetadata;
+import com.opengamma.strata.market.curve.InterpolatedNodalCurve;
+import com.opengamma.strata.market.param.CurrencyParameterSensitivities;
+import com.opengamma.strata.market.sensitivity.PointSensitivities;
+import com.opengamma.strata.market.surface.Surface;
+import com.opengamma.strata.market.surface.SurfaceMetadata;
+import com.opengamma.strata.math.impl.minimization.DoubleRangeLimitTransform;
+import com.opengamma.strata.math.impl.minimization.NonLinearTransformFunction;
+import com.opengamma.strata.math.impl.minimization.ParameterLimitsTransform;
+import com.opengamma.strata.math.impl.minimization.ParameterLimitsTransform.LimitType;
+import com.opengamma.strata.math.impl.minimization.SingleRangeLimitTransform;
+import com.opengamma.strata.math.impl.minimization.UncoupledParameterTransforms;
+import com.opengamma.strata.math.impl.statistics.leastsquare.LeastSquareResults;
+import com.opengamma.strata.math.impl.statistics.leastsquare.LeastSquareResultsWithTransform;
+import com.opengamma.strata.math.impl.statistics.leastsquare.NonLinearLeastSquare;
+import com.opengamma.strata.pricer.model.SabrParameters;
+import com.opengamma.strata.pricer.option.RawOptionData;
+import com.opengamma.strata.pricer.rate.RatesProvider;
+import com.opengamma.strata.product.capfloor.ResolvedIborCapFloorLeg;
+
+/**
+ * Caplet volatilities calibration to cap volatilities based on SABR model.
+ * <p>
+ * The SABR model parameters are computed by bootstrapping along the expiry time dimension. 
+ * The result is a complete set of curves for the SABR parameters spanned by the expiry time.  
+ * The position of the node points on the resultant curves corresponds to market cap expiries, 
+ * and are interpolated by a local interpolation scheme. 
+ * See {@link SabrIborCapletFloorletBootstrapDefinition} for detail.
+ * <p>
+ * The calibration to SABR is computed once the option volatility date is converted to prices. Thus we should note that 
+ * the error values in {@code RawOptionData} are applied in the price space rather than the volatility space.
+ */
+public class SabrIborCapletFloorletVolatilityBootstrapper extends IborCapletFloorletVolatilityCalibrator {
+
+  /**
+   * Default implementation.
+   */
+  public static final SabrIborCapletFloorletVolatilityBootstrapper DEFAULT = of(
+      VolatilityIborCapFloorLegPricer.DEFAULT, SabrIborCapletFloorletPeriodPricer.DEFAULT, 1.0e-10, ReferenceData.standard());
```

Comment:
The first pricer is used to convert the market vols to prices, whereas the second pricer is used for calibration to SABR model, that is, the second one must be SABR rather than generic `VolatilityIborCapletFloorletPeriodPricer`. In this case, 'validate(IborCapletFloorletVolatilities)' is overridden and the pricer works only when `IborCapletFloorletVolatilities` is an instance of `SabrIborCapletFloorletVolatilities`. This pricer can not be used for the vol-to-price conversion.